builtin/logical/pki/acme_state.go

```diff
@@ -118,12 +128,16 @@ func (a *ACMEState) LoadJWK(keyID string) ([]byte, error) {
 	return jwk.([]byte), nil
 }
 
-func (a *ACMEState) ParseRequestParams(data *framework.FieldData) (*JWSCtx, map[string]interface{}, error) {
-	var c JWSCtx
+func (a *acmeState) ParseRequestParams(data *framework.FieldData) (*jwsCtx, map[string]interface{}, error) {
+	var c jwsCtx
 	var m map[string]interface{}
 
 	// Parse the key out.
-	jwkBase64 := data.Get("protected").(string)
+	jwkBase64, ok := data.GetOk("protected").(string)
+	if !ok {
+		return nil, nil, fmt.Errorf("missing required field 'protected': %w", ErrMalformed)
+	}
+
 	jwkBytes, err := base64.RawURLEncoding.DecodeString(jwkBase64)
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to base64 parse 'protected': %w", err)
```

Comment:
nit: Should be wrapped with a `ErrMalformed`, with perhaps the err.Error() being %s for additional info? this and the Unmarshing error?

Not really sure what the JWS signature failure below should return, perhaps the same `ErrMalformed`?

Reply:
I was worried about the JWS Sig in particular, if there's a read call and a storage error, I wasn't sure how we wanted to handle it. Per our zoom discussion, I think wrapping storage errors with `%s` in this case would be fine, but... Not sure.